pkg/distributor/distributor.go

```diff
@@ -479,6 +479,11 @@ func (d *Distributor) createShards(stream logproto.Stream, totalShards int, tena
 		streamCount = streamCount(totalShards, stream)
 	)
 
+	if totalShards <= 0 {
+		level.Error(util_log.Logger).Log("msg", "attempt to create shard with zeroed total shards")
```

Comment:
Can you walk me through how this log message would help you isolate the problem in production?

Reply:
I'd know that somehow we're calling `createShards` with zero totalShards. By knowing that I could
- Isolate if it was introduced by a recent version (in case the log wasn't present before)
- Knowing that totalShards is zero is enough to start a deeper investigation of how it happened and I'd then enable verbose logs for this etc

Reply:
> Isolate if it was introduced by a recent version (in case the log wasn't present before)
What if we spot this only in 3 months? I'm not sure the log line contains enough context. e.g. which tenant? which stream? other shard details?

> I'd then enable verbose logs for this etc
What do you mean by this?

Reply:
OK cool. Last question: should this be an alert? Would we need to know that this happened? What's the impact?